src/ClientInterface.php

```diff
@@ -21,151 +28,108 @@ interface ClientInterface
     /**
      * Test connection.
      *
-     * @param string|null $value Any string value (server must returns it back)
+     * @param DevPingParams $params
      *
      * @throws BadRequestException
      * @throws BadResponseException
      *
      * @return DevPingResponse
      */
-    public function devPing(?string $value = null): DevPingResponse;
+    public function devPing(DevPingParams $params): DevPingResponse;
 
     /**
      * Debug token generation.
      *
-     * @param string        $username  User identifier (e.g.: `test@test`)
-     * @param string        $password  User password
-     * @param bool          $is_hash   Password hashed?
-     * @param DateTime|null $date_from Token availability start date
-     * @param int           $age       Token lifetime (in seconds)
+     * @param DevTokenParams $params
      *
      * @throws BadRequestException
      * @throws BadResponseException
      *
      * @return DevTokenResponse
      */
-    public function devToken(string $username,
-                             string $password,
-                             bool $is_hash = false,
-                             ?DateTime $date_from = null,
-                             int $age = 60): DevTokenResponse;
+    public function devToken(DevTokenParams $params): DevTokenResponse;
 
     /**
      * Retrieve information about current user.
      *
-     * @param bool $detailed
+     * @param UserParams $params
      *
      * @throws BadRequestException
      * @throws BadResponseException
      *
      * @return UserResponse
      */
-    public function user(bool $detailed = false): UserResponse;
+    public function user(UserParams $params): UserResponse;
```

Comment:
Also in this method 
```php
user(?UserParams $params)
```

Reply:
Now optional parameters are used in methods:

- `devPing()`
- `user()`
- `userReportTypes()`
- `userReports()`

Reply:
great!